src/zenml/integrations/modal/orchestrators/modal_orchestrator.py

```diff
@@ -0,0 +1,206 @@
+        """Settings class for the Modal orchestrator.
+
+        Returns:
+            The settings class.
+        """
+        from zenml.integrations.modal.flavors.modal_orchestrator_flavor import (
+            ModalOrchestratorSettings,
+        )
+
+        return ModalOrchestratorSettings
+
+    def _setup_modal_client(self) -> None:
+        """Setup Modal client with authentication."""
+        setup_modal_client(
+            token_id=self.config.token_id,
+            token_secret=self.config.token_secret,
+            workspace=self.config.workspace,
+            environment=self.config.modal_environment,
+        )
+
+    @property
+    def validator(self) -> Optional[StackValidator]:
+        """Ensures there is a container registry and artifact store in the stack.
+
+        Returns:
+            A `StackValidator` instance.
+        """
+        validator: StackValidator = create_modal_stack_validator()
+        return validator
+
+    def get_orchestrator_run_id(self) -> str:
+        """Returns the active orchestrator run id.
+
+        Raises:
+            RuntimeError: If the environment variable specifying the run id
+                is not set.
+
+        Returns:
+            The orchestrator run id.
+        """
+        try:
+            return os.environ[ENV_ZENML_MODAL_ORCHESTRATOR_RUN_ID]
+        except KeyError:
+            raise RuntimeError(
+                "Unable to read run id from environment variable "
+                f"{ENV_ZENML_MODAL_ORCHESTRATOR_RUN_ID}."
+            )
+
+    def prepare_or_run_pipeline(
+        self,
+        deployment: "PipelineDeploymentResponse",
+        stack: "Stack",
+        environment: Dict[str, str],
+        placeholder_run: Optional["PipelineRunResponse"] = None,
+    ) -> Optional[Iterator[Dict[str, MetadataType]]]:
+        """Runs the complete pipeline using Modal sandboxes.
+
+        Args:
+            deployment: The pipeline deployment to prepare or run.
+            stack: The stack the pipeline will run on.
+            environment: Environment variables to set in the orchestration
+                environment.
+            placeholder_run: An optional placeholder run for the deployment.
+
+        Raises:
+            Exception: If pipeline execution fails.
+
+        Returns:
+            None if the pipeline is executed synchronously, otherwise an iterator of metadata dictionaries.
+        """
+        if deployment.schedule:
+            logger.warning(
+                "Modal Orchestrator currently does not support the "
+                "use of schedules. The `schedule` will be ignored "
+                "and the pipeline will be run immediately."
+            )
+
+        # Setup Modal authentication
+        self._setup_modal_client()
+
+        # Generate orchestrator run ID and include pipeline run ID for isolation
+        orchestrator_run_id = str(uuid4())
+        environment[ENV_ZENML_MODAL_ORCHESTRATOR_RUN_ID] = orchestrator_run_id
```

Comment:
There is no need to pass this to the entrypoint if it is simply a random UUID. In that case, we can simply generate it in the entrypoint itself.